### PR TITLE
Pam A simple multi-threaded PAM tool to authenticate users

### PR DESCRIPTION
**Pull request by rcritten**
pam_loginuid.so needs to be disabled otherwise the session
creation will fail. This is because the thread id is different
from the pid so the kernel will reject a change to
/proc/self/loginuid

Usage is: ./thread --threads <num>

The default pam service is login. This can be overridden
with the --service option.

Signed-off-by: Rob Crittenden <rcritten@redhat.com>

### CODE REVIEW

**Comment by rcritten** [comment]
We don't have any tagged releases yet so to test this create your own archive:
```
git archive -v --format=tar --prefix=freeipa-perftest-0.1/ HEAD | gzip > 0.1.tar.gz
cp 0.1.tar.gz freeipa-perftest.spec ~/rpmbuild/SOURCES
cd ~/rpmbuild/SOURCES
rpmbuild -ba freeipa-perftest.spec
```
Then build the resulting srpm in ../SRPMS using mock, copr, whatever (for testing).

Once we have a release we can automate this a bit easier.

**Comment by antoniotorresm** [comment]
I'm thinking maybe a better option would be to package the entire repo since we are calling the package `freeipa-perftest`. That way the same package can be installed on the controller and on the clients. What do you think @rcritten?

File: src/pamtest.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        }
+    }
+    if (c != -1) {
+        poptPrintUsage(pctx, stdout, 0);
+        ret = 1;
+        goto done;
+    } 
+
+    if (threads == -1) {
+        printf("--threads is required\n");
```
Comment (antoniotorresm) [inline src/pamtest.c:127]: We could print usage here too.

**Comment by antoniotorresm** [comment]
We agreed on IRC to have two packages: one for controller and one for client. The one for the controller will require all packages currently listed on README (vagrant, ansible, etc) and it will include the scripts needed to launch `ipaperftest`. The client package will include the `pamtest` and all its dependencies.

**Comment by rcritten** [comment]
I decided to leave the C source in src/ since that is a reasonable prefix and it doesn't interfere with the python code.

File: freeipa-perftest.spec [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+URL:            https://github.com/freeipa/freeipa-perftest.git
+Source0:        https://github.com/freeipa/freeipa-perftest/archive/%{version}.tar.gz
+
+BuildRequires:  pam-devel
+BuildRequires:  krb5-devel
+BuildRequires:  popt-devel
+BuildRequires:  glibc-devel
+BuildRequires:  autoconf
+BuildRequires:  automake
+BuildRequires:  gcc
```
Comment (antoniotorresm) [inline freeipa-perftest.spec:16]: We need to add `python3-devel` to BuildRequires.

Comment (rcritten) [reply freeipa-perftest.spec:16]: Added

**Comment by antoniotorresm** [comment]
LGTM

**Pull request merged**